### PR TITLE
Registered user flow

### PR DESCRIPTION
**Pull request by matigrieben**

### CODE REVIEW

File: src/DonateTo.WebApplication/ClientApp/src/app/donation/components/donation/list/donation-list.component.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.save {
+  margin-right: 8px;
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/donation/components/donation/list/donation-list.component.css:2]: should be in em

File: src/DonateTo.ApplicationCore/Entities/DonationRequest.cs [hunk 2]
```diff
@@ -17,7 +17,7 @@ public class DonationRequest : EntityBase
         public long UserId { get; set; }
         public User User { get; set; }
         public IEnumerable<DonationRequestItem> DonationRequestItems { get; set; }
-        public DateTime FinishDate { get; set; }
+        public DateTime? FinishDate { get; set; }
```
Comment (nicolasgranata) [inline src/DonateTo.ApplicationCore/Entities/DonationRequest.cs:20]: Did you add the migration to change it to nullable?

File: src/DonateTo.WebApplication/ClientApp/src/app/donation/components/donation/list/donation-list.component.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<nz-table [nzData]="editCache" [nzLoading]="isLoading">
+  <thead>
+    <tr>
+      <th nzColumnKey="itemName" nzWidth="25%">Item</th>
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/donation/components/donation/list/donation-list.component.html:4]: The th values should use the translation service

File: src/DonateTo.WebApplication/ClientApp/src/app/donation/components/donation/list/donation-list.component.html [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      <ng-template #editTemplate>
+        <td>
+          <nz-input-number
+            [(ngModel)]="data.quantityToDonate"
+            [nzMin]="0"
+            [nzMax]="data.item.finishQuantity - data.item.currentQuantity"
+          ></nz-input-number>
+        </td>
+        <td>
+          <a (click)="saveEdit(data.id)" class="save">Save</a>
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/donation/components/donation/list/donation-list.component.html:29]: Add the translation

File: src/DonateTo.WebApplication/ClientApp/src/app/donation/donation.component.less [hunk 5]
```diff
@@ -0,0 +1,7 @@
+.donations-container {
+  padding-top: 30px;
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/donation/donation.component.less:2]: should be in em. We are not using less so this file should be css

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/async-services/http/donation-request.service.ts [hunk 6]
```diff
@@ -20,6 +20,10 @@ export class DonationRequestService extends BaseHttpClientService<DonationReques
     return this.get();
   }
 
+  getDonationRequest(id: number) {
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/async-services/http/donation-request.service.ts:23]: I think we should specify what we return

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/models/donation-request-item.model.ts [hunk 7]
```diff
@@ -6,5 +6,6 @@ export class DonationRequestItemModel extends BaseModel {
   observation: string;
   currentQuantity: number;
   finishQuantity: number;
+  unit: string;
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/models/donation-request-item.model.ts:9]: please, pull changes from develop, we need to change some entities, for example unit is not longer a string

File: src/DonateTo.WebApplication/ClientApp/src/environments/environment.ts [hunk 8]
```diff
@@ -4,7 +4,7 @@
 
 export const environment = {
   production: false,
-  baseUrl: 'https://localhost:5001',
+  baseUrl: 'https://localhost:44328',
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/environments/environment.ts:7]: Don't commit this change

File: src/DonateTo.WebApplication/ClientApp/src/app/donation/donation.component.ts [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+      this.donationSandbox.loadDonationRequest(params['donationRequestId']);
+    });
+
+    this.donationSandbox.donationRequest$.subscribe((donationRequest) => {
+      this.donation = donationRequest;
+    });
+
+    this.donationSandbox.donationRequestLoading$.subscribe((state) => {
+      this.isLoading = state;
+    });
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/donation/donation.component.ts:37]: Remove this subscription

File: src/DonateTo.WebApplication/ClientApp/src/app/donation/donation.component.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+  ngOnInit(): void {
+    this.router.params.subscribe((params: Params) => {
+      this.donationRequestId = params['donationRequestId'];
+      this.donationSandbox.loadDonationRequest(params['donationRequestId']);
+    });
+
+    this.donationSandbox.donationRequest$.subscribe((donationRequest) => {
+      this.donation = donationRequest;
+    });
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/donation/donation.component.ts:33]: Move these subscriptions to a new method called registerEvents and add the subscriptions to an Array of Subscriptions

File: src/DonateTo.WebApplication/ClientApp/src/app/donation/donation.component.ts [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+import { Subscription } from 'rxjs/internal/Subscription';
+import { DonationSandbox } from './donation.sandbox';
+import { DonationRequestModel } from '../shared/models';
+
+@Component({
+  selector: 'app-donation',
+  templateUrl: './donation.component.html',
+  styleUrls: ['./donation.component.less'],
+})
+export class DonationComponent implements OnInit {
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/donation/donation.component.ts:12]: Implement OnDestroy interface and unsubscribe from all the subscriptions.

File: src/DonateTo.WebApplication/ClientApp/src/app/home/home.component.ts [hunk 12]
```diff
@@ -61,17 +72,35 @@ export class HomeComponent implements OnInit {
...
-  goToDonate() {
-    this.homeSandbox.login();
+  goToDonate(donationRequestId: any) {
+    this.hideModal();
+    this.router.navigate(['donation', donationRequestId]);
+    // if (this.homeSandbox.isAuthenticated) {
+    //   this.hideModal();
+    //   this.router.navigate(['donation', donationRequestId]);
+    // } else {
+    //   this.homeSandbox.login();
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/home/home.component.ts:103]: For the unregistered users when they click on Donate they should be redirected to login page.

File: src/DonateTo.WebApplication/ClientApp/src/environments/environment.ts (hunk 8 above)
Comment (guillemurano) [reply src/DonateTo.WebApplication/ClientApp/src/environments/environment.ts:3]: +1

File: src/DonateTo.WebApplication/ClientApp/src/app/donation/components/donation/list/donation-list.component.html [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+            [nzMax]="data.item.finishQuantity - data.item.currentQuantity"
+          ></nz-input-number>
+        </td>
+        <td>
+          <a (click)="saveEdit(data.id)" class="save">{{
+            'RequestDonation.DonationItemList.SaveButton' | translate
+          }}</a>
+          <a
+            nz-popconfirm
+            nzPopconfirmTitle="Sure to cancel?"
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/donation/components/donation/list/donation-list.component.html:43]: Use translation service/directive/pipe

**Pull request merged**